pkg/nix/cmd/build.go

```diff
@@ -13,10 +16,30 @@ func Build(dir string, attribute string) error {
 	}
 	cmd := exec.Command("nix", "build", attribute, "-o", dir)
 
-	cmd.Stdout = os.Stdout
-	// TODO: in future- we can pipe to stderr pipe and modify error messages to be understandable by the user
-	cmd.Stderr = os.Stderr
-
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		log.Fatalf("could not get stderr pipe: %v", err)
+	}
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		log.Fatalf("could not get stdout pipe: %v", err)
+	}
+	go func() {
+		merged := io.MultiReader(stderr, stdout)
+		scanner := bufio.NewScanner(merged)
+		for scanner.Scan() {
+			msg := scanner.Text()
+			dir, err:= os.Getwd()
+			if err!=nil{
+				panic(err)
+			}
+			warning:= fmt.Sprintf("warning: Git tree '%s' is dirty", dir)
+			if msg==warning{
+				msg = fmt.Sprintf("warning: Git tree '%s' is dirty.\nThis implies you have not checked-in files in the git work tree (hint: git add)", dir)
+			}
+			fmt.Printf("%s\n", msg)
```

Comment:
I would suggest not to log/print at all in pkg/ .
This will allow us to use this package without any issues.  

When we do want our package to log, we should let the user configure the logger and other details. In this case, we don't need to.

Reply:
Will it be alright if build command also gives these messages?

![Screenshot from 2024-07-11 16-03-23](https://github.com/buildsafedev/bsf/assets/65224535/6fd06dd4-8ca1-437d-8e34-86c7309ffb73)

Reply:
yep, that's fine.